### PR TITLE
Performance improvements

### PR DESCRIPTION
**Pull request by segiddins**

### CODE REVIEW

**Comment by CocoaPodsBarista** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please include a CHANGELOG entry to credit yourself! <br />
You can find it at <a href="https://github.com/CocoaPods/CocoaPods/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
  </tbody>
</table>

Here's an example of your CHANGELOG entry:

```markdown
* Performance improvements  
  [segiddins](https://github.com/segiddins)
  [#issue_number](https://github.com/CocoaPods/CocoaPods/issues/issue_number)
```

*note*: There are two invisible spaces after the entry's text.


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by segiddins** [comment]
No functional changes, so no changelog necessary

File: lib/xcodeproj/config.rb [hunk 1]
```diff
@@ -288,11 +287,9 @@ def merge_attributes!(attributes)
       @attributes.merge!(attributes) do |_, v1, v2|
         v1 = v1.strip
         v2 = v2.strip
-        v1_split = v1.shellsplit
-        v2_split = v2.shellsplit
-        if (v2_split - v1_split).empty? || v1_split.first(v2_split.size) == v2_split
+        if v1.start_with?(v2)
           v1
-        elsif v2_split.first(v1_split.size) == v1_split
+        elsif v1.start_with?(v1)
```
Comment (DanToml) [inline lib/xcodeproj/config.rb]: I think v1 is always going to start with v1?

File: lib/xcodeproj/project/object_attributes.rb [hunk 2]
```diff
@@ -319,7 +319,7 @@ def attribute(name, klass, default_value = nil)
               if existing.is_a?(Hash) && value.is_a?(Hash)
                 return value if existing.keys == value.keys && existing == value
               elsif existing == value
-                return value if existing == value
+                return value
```
Comment (DanToml) [inline lib/xcodeproj/project/object_attributes.rb]: Lol

File: lib/xcodeproj/config.rb (hunk 1 above)
Comment (segiddins) [reply lib/xcodeproj/config.rb]: reverted this commit, we didnt have sufficient test coverage anyways :(

**Pull request merged**